python/triton/language/libdevice.py

```diff
@@ -975,10 +1000,16 @@ def rsqrt_rn(arg0, _builder=None):
 
 @extern.extern
 def ffs(arg0, _builder=None):
-    return extern.elementwise("libdevice", LIBDEVICE_PATH, [arg0, ],
-                              {(core.dtype("int32"),): ("__nv_ffs", core.dtype("int32")),
-                               (core.dtype("int64"),): ("__nv_ffsll", core.dtype("int32")),
-                               }, _builder)
+    if torch.version.hip is not None:
+        return extern.elementwise("libdevice", LIBDEVICE_PATH, [arg0, ],
+                                  {(core.dtype("int32"),): ("__ockl_ctz_u32", core.dtype("int32")),
```

Comment:
this is almost right but the semantics are different.

the bit position returned is computed starting at '0' but for __nv_ffs, it starts at '1'.

so there is an 'off-by-1' error because of the different semantics.

Reply:
now the semantic between device functions are understood, it can be fixed at python level?

Reply:
somehow i have to figure out how to add 1 to the result of the ffs

Reply:
@rsanthanam-amd wouldn't `return extern.elementwise(...) + [1]*len(arg0)` be sufficient?

Reply:
fixed this.